multiapps-controller-process/src/test/java/org/cloudfoundry/multiapps/controller/process/steps/PollServiceInProgressOperationsExecutionTest.java

```diff
@@ -23,122 +26,103 @@
 import org.cloudfoundry.multiapps.controller.process.util.ServiceOperationGetter;
 import org.cloudfoundry.multiapps.controller.process.util.ServiceProgressReporter;
 import org.cloudfoundry.multiapps.controller.process.variables.Variables;
-import org.junit.Before;
-import org.junit.Rule;
-import org.junit.rules.ExpectedException;
-import org.junit.runner.RunWith;
-import org.junit.runners.Parameterized;
-import org.junit.runners.Parameterized.Parameters;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
 import org.mockito.Mock;
 
-@RunWith(Parameterized.class)
-public class PollServiceInProgressOperationsExecutionTest extends AsyncStepOperationTest<CheckForOperationsInProgressStep> {
+class PollServiceInProgressOperationsExecutionTest extends AsyncStepOperationTest<CheckForOperationsInProgressStep> {
 
     private static final String TEST_SPACE_ID = "test";
     private static final String TEST_PROVIDER = "testProvider";
     private static final String TEST_PLAN = "testPlan";
     private static final String TEST_VERSION = "0.0.1-beta";
 
-    @Parameters
-    public static Iterable<Object[]> getParameters() {
-        return Arrays.asList(new Object[][] {
+    @Mock
+    private ServiceOperationGetter serviceOperationGetter;
+    @Mock
+    private ServiceProgressReporter serviceProgressReporter;
+    @Mock
+    private EventsGetter eventsGetter;
+    @Mock
+    private CloudControllerClient client;
+
+    private boolean shouldVerifyStepLogger;
+    private AsyncExecutionState expectedExecutionState;
+
+    public static Stream<Arguments> testPollStateExecution() {
+        return Stream.of(
 // @formatter:off
             // (0) With 2 services in progress:
-            {
-                Arrays.asList("service1","service2", "service3"), 
-                Arrays.asList(ServiceOperation.Type.DELETE, ServiceOperation.Type.CREATE, ServiceOperation.Type.DELETE),
-                Arrays.asList(ServiceOperation.State.IN_PROGRESS, ServiceOperation.State.SUCCEEDED, ServiceOperation.State.IN_PROGRESS), 
-                false, AsyncExecutionState.RUNNING, null
-            },
+            Arguments.of(Arrays.asList("service1","service2", "service3"),
```

Comment:
Here are a lot of `Arrays.asList`

Reply:
I've replaced them wherever was possible but there are several occurrences where null is passed in asList method and this cannot be replaced because List.of requires non-null objects.